### PR TITLE
Monasca: fix agent is not ready in each barclamp

### PR DESCRIPTION
**Pull request by ilausuch**
The problem: When the barclamps are preparing the monasca monitor
it cannot find a resource matching openstack-monasca-agent.target
because is not defined.

The solution: Delay the monitoring until the agent is already
defined. To do that the monitor setup is performed also inside of
monasca agent recipe. Additionally the monitor setup has a check
to be sure the monasca master, service and agent are running.

### CODE REVIEW

File: chef/cookbooks/crowbar-openstack/libraries/helpers.rb [hunk 1]
```diff
@@ -255,6 +255,14 @@ def self.rabbitmq_settings(node, barclamp)
     @rabbitmq_settings[instance]
   end
 
+  def self.no_monasca_server_or_master(node)
```
Comment (jgrassler) [inline chef/cookbooks/crowbar-openstack/libraries/helpers.rb]: There's no need to pass a node object to this function, for...

File: chef/cookbooks/crowbar-openstack/libraries/helpers.rb [hunk 2]
```diff
@@ -255,6 +255,14 @@ def self.rabbitmq_settings(node, barclamp)
...
+    monasca_master = CrowbarUtilsSearch.node_search_with_cache(node, "roles:monasca-master").first
```
Comment (jgrassler) [inline chef/cookbooks/crowbar-openstack/libraries/helpers.rb]: ...for you can search for the Monasca node without one as we do here:

https://github.com/crowbar/crowbar-openstack/blob/master/chef/cookbooks/monasca/recipes/agent.rb#L24

Comment (jgrassler) [reply chef/cookbooks/crowbar-openstack/libraries/helpers.rb]: First option sounds more reasonable, yes.

File: chef/cookbooks/crowbar-openstack/libraries/helpers.rb [hunk 3]
```diff
@@ -32,6 +32,14 @@ def fetch_database_connection_string(db_auth, barclamp = @cookbook_name)
     def fetch_rabbitmq_settings(barclamp=@cookbook_name)
       CrowbarOpenStackHelper.rabbitmq_settings(node, barclamp)
     end
+
+    def no_monasca_server_or_master
+      monasca_master = node_search_with_cache("roles:monasca-master").first
+      return true if monasca_master.nil?
```
Comment (cmurphy) [inline chef/cookbooks/crowbar-openstack/libraries/helpers.rb]: (nitpick) you could replace this with `return monasca_master.nil?`

Comment (ilausuch) [reply chef/cookbooks/crowbar-openstack/libraries/helpers.rb]: Done

Comment (ilausuch) [reply chef/cookbooks/crowbar-openstack/libraries/helpers.rb]: I put it in the second one

Comment (ilausuch) [reply chef/cookbooks/crowbar-openstack/libraries/helpers.rb]: @cmurphy you were wright. I change it in my last commit

**Pull request merged**